### PR TITLE
Create embedded schema guide

### PR DESCRIPTION
**Pull request by matteing**
Hi folks,

Learning to use embedded schemas was a struggle for me as an Elixir beginner. There isn't much documentation out there, especially aimed at beginners looking for practical usages of this awesome feature. 

Due to this situation, I decided to create a small, pragmatic guide on how to use embedded schemas in a simple application.  This is my first contribution to anything Elixir-related, so feedback is very welcome. 

[Here's a rendered preview of the guide.](https://gist.github.com/matteing/d4349860a3bc1facbff706dfad00749a)

### CODE REVIEW

**Comment by matteing** [comment]
> Hey :)
> 
> I think embeds_one/3 currently provide all the info there plus a little extra, and the "Data mapping and validation" guide does the introduction of the concept.
> 
> But maybe its a good idea to have it separate as well.

Hey! You're right, yes: all the information for this is there, but it is a little sparse. I propose this more of a unifying "recipe": my greatest difficulty was connecting all the dots.

I recall wondering "do I need a migration for this?", then wondering "how do I write it?" then Googling for quite a bit to find the answer in a blog. There wasn't a guide specific to this feature that put all the steps together (I found the schemas documentation page a little hard to read for an Elixir beginner).

This is a more pragmatic, unified, slightly opinionated guide to make sure newbies like me have a place to begin exploring the feature. :)

**Comment by josevalim** [comment]
Hi @matteing, thanks for the PR! I think having this in a unified place is great. I believe it would be even more useful if we convert it to an actual introductory guide and also explain what are embeds and so on, and include some examples of using them in the command line (similar to the actual Getting Started guide). So what do you think about this:

1. Let's call this "Embedded schemas", move it to the Introduction part, and have it after "Getting Started"
2. Add some introduction to the chapter, explaining embedded changesets, why would someone use them, etc
3. Include some examples of invoking and casting the changeset through the parent
4. If you want, even include some wording in the migration section about how they are represented in the database
5. Bonus points if you want to show how embedded queries work: 

    from u in User, where: u.user_profile.dark_mode == true

I think with those we will have a new and fantastic guide on embeds! WDYT?

> EDIT: I realize it might be even better to add a small note about migrations within the embeds_one/3 documentation as well, I can go ahead and do that if you give me the green light!

Yes, we should add this regardless. :)

File: guides/howtos/Using embedded schemas.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+In this short recipe, we'll learn how to create a basic embedded schema within an existing Ecto schema.
+
+## Scenario
+
+Imagine this scenario: we're building an application that needs to store user profile information. We'd have a `User` schema that looks like this:
+
+```elixir
+# User schema
+schema "users" do
+    field :is_active, :boolean
```
[the remark this answers is not in the document]
Comment (matteing) [reply guides/howtos/Using embedded schemas.md:12]: Sounds good! I'll fix the indent + make the fields more consistent.

**Comment by aegatlin** [comment]
Hey all, excited to see this go live. I noticed it's been almost two months since a commit to this PR. @matteing , any updates on progress, or anything we can do to help? I love the introduction content. It's exactly what I want as an intro to embedded schemas

PS As a person with chronic arm pain, I def hope you take care of your hand health first! <3

**Comment by matteing** [comment]
Hey @aegatlin, thanks for checking up. Quite honestly, I haven’t been able to pick this up - school and freelance work have me pinned at the moment. I might have some free time by end of December, but I don’t want to promise anything. I personally haven’t been able to write any Elixir in the past couple of weeks.

Commits to this pull are welcome though - I’d love to see some progress to bring this to the docs!

**Comment by aegatlin** [comment]
My initial, incorrect, idea for how to contribute here was to fork @matteing 's fork, and then submit a PR to that fork, which when accepted would add the commits downstream to here. It seems this is not how forks work, and I can't do that.

I'm naively thinking one way to do this would be creating a branch for the work done so far. That way I can submit PRs against that branch directly. What do you think @josevalim ? I can think of a few other ways to do it as well but I haven't been in this situation before, so keen to hear suggestions for how best to contribute. Thanks! :D

**Comment by josevalim** [comment]
@aegatlin I will do my best to review what we have so far and merge it without the TODO sections. And when you can send PRs for each of the TODO sections? :)

**Comment by aegatlin** [comment]
Do you mean to merge it directly to master, or to put it on a branch? If you want to merge it into master, maybe you can wait until I let you know here that I have written up the rest of the doc, that way I can quickly follow up your merge with my own PR. How does that sound? It would take a few days. :)

**Comment by josevalim** [comment]
@aegatlin i will merge to master today, yeah. I think we have enough to get some foundation working.

**Comment by aegatlin** [comment]
I will begin working on the additional details inspired by the remaining todos and your bullet points from above. How exciting! Thanks for kicking this all off @matteing :D

File: guides/howtos/Using embedded schemas.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# Using embedded schemas
+
+In this short recipe, we'll learn how to create a basic embedded schema within an existing Ecto schema.
```
Comment (josevalim) [inline guides/howtos/Using embedded schemas.md:3]:
```suggestion
In this short how-to, we'll learn how to create a basic embedded schema within an existing Ecto schema.
```

File: guides/introduction/Embedded Schemas.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+```elixir
+defmodule User do
+  use Ecto.Schema
+
+  schema "users" do
+    field :is_active, :boolean
+    field :email, :string
+    field :confirmed_at, :naive_datetime
+
+
```
Comment (josevalim) [inline guides/introduction/Embedded Schemas.md:33]:
```suggestion
```

File: guides/introduction/Embedded Schemas.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    timestamps()
+  end
+end
+```
+
+Let's kick things off with a tutorial to explain how to recreate this use case: building a solution for user profiles in an example app.
+
+### Writing the schema
+
+The first step is to write the structure of the data we're storing. In the case of our profile, we'd like to store the user's age, gender and a profile picture.
```
Comment (josevalim) [inline guides/introduction/Embedded Schemas.md:49]:
```suggestion
The first step is to write the structure of the data we're storing. In the case of our profile, we'd like to store the user's age, favorite color, and a profile picture.
```

File: guides/introduction/Embedded Schemas.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+  schema "users" do
+    field :is_active, :boolean
+    field :email, :string
+    field :confirmed_at, :naive_datetime
+
+
+    embeds_one :profile do
+      field :age, :integer
+      field :gender, :integer
```
Comment (josevalim) [inline guides/introduction/Embedded Schemas.md:67]:
```suggestion
      field :favorite_color, Ecto.Enum, values: [:red, :green, :blue, :pink, :black, :orange]
```

File: guides/introduction/Embedded Schemas.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+    timestamps()
+  end
+end
+
+# user/user_profile.ex
+defmodule UserProfile do
+  embedded_schema "profile" do
+    field :age, :integer
+    field :gender, :integer
```
Comment (josevalim) [inline guides/introduction/Embedded Schemas.md:99]:
```suggestion
    field :favorite_color, Ecto.Enum, values: [:red, :green, :blue, :pink, :black, :orange]
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**